projects/jupyter-server-ydoc/jupyter_server_ydoc/app.py

```diff
@@ -134,6 +138,33 @@ def initialize_handlers(self):
             ]
         )
 
+    async def start_extension(self):
+        """Start the y-websocket server.
+        """
+        self.log.info("Starting the Collaborative Document Server.")
+
+        def _restart_or_teardown_yserver(_):
+            """If the y websocket server task stopped due to an exception, restart it.
+            
+            If the y-websocket server was cancelled on purpose, tear it down.
+            """
+            self.ywebsocket_server._started = None
+            self.ywebsocket_server._starting = False
```

Comment:
This attribute doesn't exist anymore after https://github.com/jupyter-server/pycrdt-websocket/pull/24 (which was not released yet, but it will be soon).

Reply:
FYI it was released in `pycrdt-websocket` v0.13.0.